libs/langchain/langchain/retrievers/web_research.py

```diff
@@ -195,6 +206,9 @@ def _get_relevant_documents(
         # Check for any new urls that we have not processed
         new_urls = list(urls.difference(self.url_database))
 
+        # Removes urls with connection errors from list
+        new_urls = self.fetch_valid_documents(new_urls)
```

Comment:
this will significantly slow down execution. 

I think the best option would be a combination approach of what you did here and in #13353, where you add a config param to `AsyncHtmlLoader` that defaults to False of `ignore_load_errors`, and you construct the loader here with that set to true.

What do you think?

Reply:
It does make sense to me. I've done the `try ... except ...` in the `_scrape` method to avoid putting a lot of logic in the `load` method.